src/basic/TRAP_INT-Hip.cpp

```diff
@@ -0,0 +1,160 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-19, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "TRAP_INT.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_HIP)
+
+#include "common/HipDataUtils.hpp"
+
+#include <iostream>
+
+namespace rajaperf
+{
+namespace basic
+{
+
+//
+// Function used in TRAP_INT loop.
+//
+RAJA_INLINE
+RAJA_DEVICE
+Real_type trap_int_func(Real_type x,
+                        Real_type y,
+                        Real_type xp,
+                        Real_type yp)
+{
+   Real_type denom = (x - xp)*(x - xp) + (y - yp)*(y - yp);
+   denom = 1.0/sqrt(denom);
+   return denom;
+}
+
+
+  //
+  // Define thread block size for HIP execution
+  //
+  const size_t block_size = 256;
+
+
+#define TRAP_INT_DATA_SETUP_HIP \
+  Real_type x0 = m_x0; \
+  Real_type xp = m_xp; \
+  Real_type y = m_y; \
+  Real_type yp = m_yp; \
+  Real_type h = m_h;
+
+#define TRAP_INT_DATA_TEARDOWN_HIP // nothing to do here...
+
+
+__global__ void trapint(Real_type x0, Real_type xp,
+                        Real_type y, Real_type yp,
+                        Real_type h,
+                        Real_ptr sumx,
+                        Index_type iend)
+{
+  HIP_DYNAMIC_SHARED( Real_type, psumx)
+
+  Index_type i = blockIdx.x * blockDim.x + threadIdx.x;
+
+  psumx[ threadIdx.x ] = 0.0;
+  for ( ; i < iend ; i += gridDim.x * blockDim.x ) {
+    Real_type x = x0 + i*h;
+    Real_type val = trap_int_func(x, y, xp, yp);
+    psumx[ threadIdx.x ] += val;
+  }
+  __syncthreads();
+
+  for ( i = blockDim.x / 2; i > 0; i /= 2 ) {
+    if ( threadIdx.x < i ) {
+      psumx[ threadIdx.x ] += psumx[ threadIdx.x + i ];
+    }
+     __syncthreads();
+  }
+
+#if 1 // serialized access to shared data;
+  if ( threadIdx.x == 0 ) {
+    RAJA::atomic::atomicAdd<RAJA::atomic::hip_atomic>( sumx, psumx[ 0 ] );
```

Comment:
The `atomic` namespace has been removed.
```suggestion
    RAJA::atomicAdd<RAJA::hip_atomic>( sumx, psumx[ 0 ] );
```